src/main/java/duke/command/Add.java

```diff
@@ -0,0 +1,175 @@
+package duke.command;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.TaskList;
+import duke.exception.CommandNotDefinedException;
+import duke.exception.DukeException;
+import duke.exception.EmptyTextException;
+import duke.exception.TimeFormatException;
+import duke.exception.TimeInconsistException;
+import duke.exception.WrongUsageException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Class for adding a task into task list
+ */
+public class Add implements Command {
+    static final int TODO_LENGTH = 5;
+    static final int DEADLINE_LENGTH = 9;
+    static final int BY_LENGTH = 3;
+    static final int EVENT_LENGTH = 6;
+    static final int FROM_LENGTH = 5;
+    static final int TO_LENGTH = 3;
+    static final int DEADLINE_TOKEN_NUM = 2;
+    static final int EVENT_TOKEN_NUM = 3;
+
+    private Task task;
+    private TaskList tasks;
+
+    /**
+     * Constructor
+     * add task to task list.
+     * Identify which type of task is it.
+     * @param text the description of a task
+     * @throws DukeException wrong inputs might happens
+     */
+    public Add(String text, TaskList taskList) throws DukeException {
```

Comment:
Would it be good if the variable `text` be given a more descriptive name? From what I understand it seems to refer to the entire line of command, so maybe something like `commandText` will be more descriptive.